src/lib/util.ts

```diff
@@ -13,43 +21,67 @@ const polyfills = {
 
 export const btoa = globalThis.btoa ?? polyfills.btoa;
 
-export const contractAddresses: { [index: string]: string } = {
-  // If building locally you can put your contract address here or use the contract connection option
-  localhost: "",
-  goerli: "0xa4b0729f02C6dB01ADe92d247b7425953d1DbA25",
-  "optimism-kovan": "0xf9C3530C03D335a00163382366a72cc1Ebbd39fF",
-  "polygon-mumbai": "0x70364D26743851d4FE43eCb065811402D06bf4AD",
-};
+export type NetworkName = "testnet" | "staging" | "custom";
+
+export type ChainName =
+  | "ethereum-goerli"
+  | "optimism-kovan"
+  | "polygon-mumbai"
+  | "optimism-kovan-staging"
+  | "custom";
+
+export interface SupportedChain {
+  // Matches naming convention used in https://chainlist.org and ethers network lib
+  name: string;
+  phrase: string;
+  chainId: number;
+  contract: string;
+}
 
-export const SUPPORTED_NETWORKS: SupportedNetwork[] = [
-  {
-    key: "goerli",
+export const SUPPORTED_CHAINS: Record<ChainName, SupportedChain> = {
+  // "testnet"
+  "ethereum-goerli": {
     name: "goerli",
     phrase: "Ethereum Goerli",
     chainId: 5,
+    contract: "0xa4b0729f02C6dB01ADe92d247b7425953d1DbA25",
   },
-  {
-    key: "optimism-kovan",
+  "optimism-kovan": {
     name: "optimism-kovan",
     phrase: "Optimism Kovan",
     chainId: 69,
+    contract: "0xf9C3530C03D335a00163382366a72cc1Ebbd39fF",
   },
-  {
-    key: "polygon-mumbai",
+  "polygon-mumbai": {
     name: "maticmum",
     phrase: "Polygon Testnet",
     chainId: 80001,
+    contract: "0x70364D26743851d4FE43eCb065811402D06bf4AD",
   },
-  {
-    key: "unknown",
+  // staging:
+  "optimism-kovan-staging": {
+    name: "optimism-kovan",
+    phrase: "Optimism Kovan",
+    chainId: 69,
+    contract: "0x322F01e81c38B4211529f334864fA630F6aeA408",
+  },
+  custom: {
```

Comment:
this is just a generic placeholder for any custom network, be it localhost or otherwise